Mathlib/AlgebraicGeometry/Restrict.lean

```diff
@@ -365,6 +365,16 @@ theorem Γ_map_morphismRestrict {X Y : Scheme.{u}} (f : X ⟶ Y) (U : Opens Y) :
   rw [Scheme.Γ_map_op, morphismRestrict_app f U ⊤, f.naturality_assoc, ← X.presheaf.map_comp]
   rfl
 
+@[simp]
+lemma morphismRestrict_id {X : Scheme.{u}} (U : Opens X) : 𝟙 X ∣_ U = 𝟙 _ := by
+  ext1
+  · ext; erw [morphismRestrict_val_base (𝟙 X) U]; rfl
+  · simp only [Scheme.restrict_presheaf_obj, Scheme.id_val_base, Opens.carrier_eq_coe,
+      TopCat.coe_id, id_eq, eq_mpr_eq_cast, morphismRestrict_app', Scheme.Hom.appLE, Scheme.id_app,
+      Category.id_comp, eqToHom_op, Scheme.restrict_presheaf_map, ← Functor.map_comp]
+    rw [← X.presheaf.map_id]
```

Comment:
This line is responsible for all the regression in the file. Tracing it crashes the LSP.

Reply:
I changed it into a new proof that should be a lot faster. Let's see how the new benchmark says.

Reply:
Great! It certainly looks much cleaner